### PR TITLE
Fix things that walk tokens to properly handle templates, etc.

### LINKED ISSUE #357: Backtick in regexp causes following comment rule to be ignored
If a regular expression literal containing a backtick character preceeds a comment rule then that rule is ignored, for example:

```
var  re;
re = /`/;   // Backtick in regexp causes following 'disable:no-unused-variable' rule to be ignoredj.

/* tslint:disable:no-unused-variable */
var s = 'xxx';
```

Running tslint on the above file generates _unused variable_ error:

```
$ tslint --version
2.1.1

$ tslint -f t.ts
t.ts[5, 5]: unused variable: 's'
```

### PR DESCRIPTION
**Pull request by gscshoyru**
Fixes #357, #349, #332, and possibly a couple others I missed

### CODE REVIEW

**Comment by ashwinr** [comment]
failing builds... :(

**Comment by gscshoyru** [comment]
Yup, forgot to actually include the new walker. Added it and repushed, the build is running again.

File: lib/tslint.d.ts [hunk 1]
```diff
@@ -164,7 +164,19 @@ declare module Lint {
     function findFormatter(name: string, formattersDirectory?: string): any;
 }
 declare module Lint {
-    class EnableDisableRulesWalker extends Lint.RuleWalker {
+    class SkipableTokenAwareRuleWalker extends RuleWalker {
```
Comment (adidahiya) [inline lib/tslint.d.ts]: let's spell it `SkippableTokenAwareRuleWalker`

Comment (ashwinr) [inline lib/tslint.d.ts]: +1

Comment (gscshoyru) [inline lib/tslint.d.ts]: Ok, renamed and pushed.

**Comment by ashwinr** [comment]
is the build taking much longer than usual?

**Pull request merged**